pkg/agent/informant.go

```diff
@@ -254,6 +254,40 @@ func NewInformantServer(
 		}
 	})
 
+	healthCheckerName := fmt.Sprintf("InformantServer health-checker (%s)", server.desc.AgentID)
+	runner.spawnBackgroundWorker(backgroundCtx, healthCheckerName, func(c context.Context) {
+		// FIXME: make this duration configurable
+		ticker := time.NewTicker(5 * time.Second)
+		for {
+			select {
+			case <-c.Done():
+				return
+			case <-ticker.C:
+			}
+
+			// If we've already registered with the informant, and it doesn't support health checks,
+			// exit.
+			var done bool
+			func() {
+				server.requestLock.Lock()
+				defer server.requestLock.Unlock()
+
+				if server.protoVersion != nil && !server.protoVersion.AllowsHealthCheck() {
+					runner.logger.Infof("Aborting future informant health checks because it does not support them")
+					done = true
+					return
+				}
+
+				if _, err := server.HealthCheck(c); err != nil {
+					runner.logger.Warningf("Informant health check failed: %w", err)
```

Comment:
I'm not sure, but I think the `%w` verb is `fmt.Errorf` specific.

Reply:
Yup - fixed